lib/vrl/stdlib/src/parse_json.rs

```diff
@@ -162,5 +280,41 @@ mod tests {
                 .add_array(Collection::from_unknown(inner_kind()))
                 .add_object(Collection::from_unknown(inner_kind())),
         }
+
+        max_depth {
+            args: func_args![ value: r#"{"top_layer": {"layer_one": "finish", "layer_two": 2}}"#, max_depth: 1],
+            want: Ok(value!({ top_layer: r#"{"layer_one": "finish", "layer_two": 2}"# })),
+            tdef: type_def(),
+        }
+
+        max_depth_array {
+            args: func_args![ value: r#"[{"top_layer": {"next_layer": ["finish"]}}]"#, max_depth: 2],
+            want: Ok(value!([{ top_layer: r#"{"next_layer": ["finish"]}"# }])),
```

Comment:
I am also a little confused about the layer counting here. Should this output two layers:
```suggestion
            args: func_args![ value: r#"[{"top_layer": {"next_layer": ["finish"]}}]"#, max_depth: 2],
            want: Ok(value!([{ top_layer: {"next_layer": r#"["finish"]"# }}])),
```

Reply:
In this example, the first layer is the array, and the second one is the object. I can swap them to make it more obvious or add a comment.

Reply:
Ah, okay, that's what I was missing.